designs/multi-cluster.md

```diff
@@ -94,17 +100,32 @@ logic.
 ## Proposal
 
 The `ctrl.Manager` _SHOULD_ be extended to get an optional `cluster.Provider` via
-`ctrl.Options` implementing
+`ctrl.Options`, implementing:
 
 ```golang
 // pkg/cluster
+
+// Provider defines methods to retrieve clusters by name. The provider is
+// responsible for discovering and managing the lifecycle of each cluster.
+//
+// Example: A Cluster API provider would be responsible for discovering and
+// managing clusters that are backed by Cluster API resources, which can live
+// in multiple namespaces in a single management cluster.
 type Provider interface {
-   Get(ctx context.Context, clusterName string, opts ...Option) (Cluster, error)
-   List(ctx context.Context) ([]string, error)
-   Watch(ctx context.Context) (Watcher, error)
+	// Get returns a cluster for the given identifying cluster name. Get
+	// returns an existing cluster if it has been created before.
+	Get(ctx context.Context, clusterName string) (Cluster, error)
 }
 ```
+
+A cluster provider is responsible for constructing a `cluster.Cluster` instance
+upon calls to `Get(ctx, clusterName)` and returning it. Providers should keep track
+of created clusters and return them again if the same name is requested. Since
+providers are responsible for constructing the `cluster.Cluster` instance, they
+can make decisions about e.g. reusing existing informers.
+
```

Comment:
```suggestion
A cluster provider is responsible for constructing `cluster.Cluster` instances and
returning them upon calls to `Get(ctx, clusterName)`. Providers should keep track
of created clusters and return them again if the same name is requested. Since
providers are responsible for constructing the `cluster.Cluster` instance, they
can make decisions about e.g. reusing existing informers.
```

Aka it is not necessary to construct them during the `Get` call. That's one option. But you can also construct them out-of-band, and just do a look-up on `Get`.

Reply:
That is true. Changed it.